terraform/node_output.go

```diff
@@ -293,6 +363,40 @@ func (n *NodeDestroyableOutput) Execute(ctx EvalContext, op walkOperation) error
 	if state == nil {
 		return nil
 	}
+
+	// if this is a root module, try to get a before value from the state for
+	// the diff
+	before := cty.NullVal(cty.DynamicPseudoType)
+	mod := state.Module(n.Addr.Module)
+	if n.Addr.Module.IsRoot() && mod != nil {
+		for name, o := range mod.OutputValues {
+			if name == n.Addr.OutputValue.Name {
+				before = o.Value
```

Comment:
Thanks, I missed that one! It's not required, since we're not going to find another root output with the same name, but there is no reason to _not_ break here since we found the value we're looking for!